packages/core/src/loaders/progressiveLoader.ts

```diff
@@ -0,0 +1,382 @@
+import type {
+  IRetrieveConfiguration,
+  IImage,
+  RetrieveStage,
+  EventTypes,
+  ProgressiveListener,
+} from '../types';
+import sequentialRetrieveConfiguration from './sequentialRetrieveConfiguration';
+import interleavedRetrieveConfiguration from './interleavedRetrieveConfiguration';
+import { loadAndCacheImage, loadImage } from './imageLoader';
+import { triggerEvent, ProgressiveIterator, decimate } from '../utilities';
+import imageLoadPoolManager from '../requestPool/imageLoadPoolManager';
+import { ImageStatus, RequestType, Events } from '../enums';
+import cache from '../cache';
+import eventTarget from '../eventTarget';
+
+export { sequentialRetrieveConfiguration, interleavedRetrieveConfiguration };
+
+type StageStatus = {
+  stageId: string;
+  startTime?: number;
+  stageStartTime?: number;
+  numberOfImages: number;
+  numberOfFailures: number;
+  remaining: number;
```

Comment:
this is not consistent naming, two of them has number in the begining and the last one (remaining) does not have and all are representing a count

how about 

```js
numberOfTotalImages
numberOfFailedImages
numberOfPendingImages
```

or 

```js
totalImagesCount
imageLoadFailedCount
imageLoadPendingCount
```